### ISSUE TITLE
Rustup-init minimal profile installs default profile items

### ISSUE THREAD

**Issue by deadwanderer** [rust-lang/rustup#4117]
### Verification

- [x] I searched for recent similar issues at https://github.com/rust-lang/rustup/issues?q=is%3Aissue+is%3Aopen%2Cclosed and found no duplicates.
- [x] I am on the latest version of Rustup according to https://github.com/rust-lang/rustup/tags and am still able to reproduce my issue.

### Problem

I'm installing the latest rustup-init on my Windows 11 machine (freshly downloaded from the site today), specifying `--profile minimal`, and it installs the `default`profile items (`rust-docs`, `rustfmt`, and `clippy`) as well.

Rust-docs takes over half an hour to download (projected, I don't let it run for more than 5 minutes of 8 KB/s progress before rage-quitting), so I desperately want to skip it

### Steps

1. Run `rustup-init --profile minimal`
2. Observe the purported `minimal` profile installing extra items...

### Rustup version

```console
rustup-init 1.27.1 (54dd3d00f 2024-04-24)
```

### Installed toolchains

```console
None, this is happening during/before the install
```

### OS version

```console
Windows 11 Enterprise 24H2 (26100.2314)
```

**Comment by ChrisDenton** [comment]
Hm, I wasn't able to reproduce this issue.

- I used Windows Sandbox so I had a fresh environment.
- I downloaded rustup from rustup.rs.
- I ran `./rustup-init --profile minimal`
- `profile: minimal` was shown in the displayed installation options
- It downloaded `cargo` `rust-std` and `rustc` only.

Is it possible there's something in your environment or maybe a previous installation?

**Comment by deadwanderer** [comment]
It does appear that I had something cached in my profile. I deleted the `.rustup` and `.cargo` folders and re-ran the install, and everything worked correctly. Thanks!

**Comment by rami3l** [comment]
I believe this is a duplicate of #4054.

**Comment by rami3l** [comment]
@djc Just a question: why did you open this issue again? Did you encounter any problems with the new duplicate-linking feature, or did you disagree with my triage conclusion?

**Comment by djc** [comment]
I like keeping issues in the proper closed state; you closed it in completed/done state while you mentioned it was a duplicate (and we didn't make any changes to fix it), so the invalid/duplicate state is more accurate, right?

**Comment by rami3l** [comment]
> I like keeping issues in the proper closed state; you closed it in completed/done state while you mentioned it was a duplicate (and we didn't make any changes to fix it), so the invalid/duplicate state is more accurate, right?

@djc In that case, you might be using an outdated client? I saw it having been closed in the invalid/duplicate state on my side... I'll try it again.

![Image](https://github.com/user-attachments/assets/817ce48f-4eaa-44ef-b45d-f1ffa1e61898)

**Comment by djc** [comment]
I was doing it on mobile, where it (and again -- though not after I reclosed it) is showing in the purple state:

![Image](https://github.com/user-attachments/assets/7a14038b-3408-4f1e-bd07-fcf5703b517a)

After I reclosed it showed in grey state, with a different icon. So something is iffy here...

Sorry for the noise, had not encountered this problem before! It looks like they might have a new actually "duplicated" state which maybe the (official) iOS client (the latest version according to the App Store) doesn't understand yet? Because I hadn't seen this widget before:

<img width="267" alt="Image" src="https://github.com/user-attachments/assets/c3d114d1-423c-411c-9830-d1c3a7ab13ea" />
